src/Entity/EntityIdValue.php

```diff
@@ -31,8 +31,8 @@ public function __construct( EntityId $entityId ) {
 	 */
 	public function serialize() {
 		return json_encode( [
-			$this->entityId->getEntityType(),
-			$this->getNumericId()
+			get_class( $this->entityId ),
```

Comment:
Yes, it's not pretty. But all we use this for is cloning statements.

The change is here because the old code assumes two things: a) the ID can be represented as a number (it can't) and b) we have a well known set of entity types (we don't). 

So, having the choice to try and access some kind of entity id type registry via a global, or putting the class name into the serialization, I opted for the latter.